src/modules/string/String.cpp

```diff
@@ -129,16 +162,53 @@ std::string String::hexadecimal(unsigned int length, HexCasing casing, HexPrefix
     return hexadecimal;
 }
 
-std::string String::binary(unsigned int length)
+std::string String::binary(GuaranteeMap&& guarantee, unsigned int length)
 {
-    std::string binary{"0b"};
-
-    for (unsigned i = 0; i < length; i++)
+    std::set<char> targetCharacters{'0', '1'};
+    // throw if guarantee is invalid
+    if (!isValidGuarantee(guarantee, targetCharacters, length))
     {
-        binary += static_cast<char>(Number::integer(1));
+        throw std::invalid_argument{"Invalid guarantee."};
     }
 
-    return binary;
+    std::string binary{};
+    binary += generateAtleastString(guarantee);
+    // string with least required chars cannot be greater than the total length
+    assert(binary.size() <= length);
+    // we will generate chars for remaining length only
+    length -= binary.size();
+    for (unsigned i = 0; i < length; ++i)
+    {
+        char generatedChar;
+        // generate chars till we find a usable char
+        while (true)
+        {
+            // pick random char from targetCharacters
+            std::mt19937 gen(std::random_device{}());
+            std::sample(targetCharacters.begin(), targetCharacters.end(), &generatedChar, 1, gen);
```

Comment:
replace with Helper::arrayElement<char>(targetCharacters)

Reply:
This isn't working because Helper::arrayElement() cannot take std::set as argument
I think we should have an implementation for taking random element from a std::set too as string guarantees will work using std::set only

Reply:
I can fix this once #306 is resolved